src/error.rs

```diff
@@ -1,11 +1,19 @@
+
 use defines::AfError;
 use std::error::Error;
+use std::sync::Mutex;
 
 pub type ErrorCallback = Fn(AfError);
 
 pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
```

Comment:
@9prady9 : I don't think this does what you are expecting.
I would assume this would look like [havent tested] :
```rust
lazy_static! {
    static ref ARRAY: Mutex<ErrorCallback> = Mutex::new(&handle_error_general);
}
```

Then every call would be  `HANDLE_ERROR.lock().unwrap()(af_err)`

This would ensure that when you swap out the `HANDLE_ERROR` function from a thread it would then lock the currently stored function (and thus ensure that the currently registered one would be called).